beam_jobs/base_model_validator.py

```diff
@@ -0,0 +1,239 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+import datetime
+import re
+
+import apache_beam as beam
+
+from core.domain import cron_services
+from core.platform import models
+import python_utils
+
+
+
+(base_models, user_models) = models.Registry.import_models(
+    [models.NAMES.base_model, models.NAMES.user])
+datastore_services = models.Registry.import_datastore_services()
+
+
+ERROR_CATEGORY_COMMIT_CMD_CHECK = 'commit cmd check'
+ERROR_CATEGORY_COMMIT_STATUS_CHECK = 'post commit status check'
+ERROR_CATEGORY_COUNT_CHECK = 'count check'
+ERROR_CATEGORY_CURRENT_TIME_CHECK = 'current time check'
+ERROR_CATEGORY_DATETIME_CHECK = 'datetime check'
+ERROR_CATEGORY_DOMAIN_OBJECT_CHECK = 'domain object check'
+ERROR_CATEGORY_EMAIL_CHECK = 'email check'
+ERROR_CATEGORY_ERROR_CHECK = 'error check'
+ERROR_CATEGORY_FIELD_CHECK = 'field check'
+ERROR_CATEGORY_FIRST_PUBLISHED_MSEC_CHECK = 'first published msec check'
+ERROR_CATEGORY_ID_CHECK = 'id check'
+ERROR_CATEGORY_LAST_UPDATED_CHECK = 'last updated check'
+ERROR_CATEGORY_LENGTH_CHECK = 'length check'
+ERROR_CATEGORY_NAME_CHECK = 'name check'
+ERROR_CATEGORY_OUTPUT_CHECK = 'output check'
+ERROR_CATEGORY_PRIVATE_COMMIT_CHECK = 'post commit is private check'
+ERROR_CATEGORY_PROPERTY_FETCH_CHECK = 'fetch properties'
+ERROR_CATEGORY_RATED_ON_CHECK = 'rated on check'
+ERROR_CATEGORY_RATINGS_CHECK = 'ratings check'
+ERROR_CATEGORY_REFERENCE_CHECK = 'reference check'
+ERROR_CATEGORY_AUTHOR_CHECK = 'author check'
+ERROR_CATEGORY_REVIEWER_CHECK = 'reviewer check'
+ERROR_CATEGORY_STATE_NAME_CHECK = 'state name check'
+ERROR_CATEGORY_SUMMARY_CHECK = 'summary check'
+ERROR_CATEGORY_TIME_FIELD_CHECK = 'time field relation check'
+ERROR_CATEGORY_TYPE_CHECK = 'type check'
+ERROR_CATEGORY_VERSION_CHECK = 'version check'
+ERROR_CATEGORY_STALE_CHECK = 'stale check'
+ERROR_CATEGORY_INVALID_IDS_IN_FIELD = 'invalid ids in field'
+
+VALIDATION_MODE_NEUTRAL = 'neutral'
+VALIDATION_MODE_STRICT = 'strict'
+VALIDATION_MODE_NON_STRICT = 'non-strict'
+
+
+class ValidateModelIdWithRegex(beam.DoFn):
+    """DoFn to validate model ids against a given regex string."""
+
+    def __init__(self, regex_string):
+        """Initializes the ValidateModelIdWithRegex DoFn.
+
+        Args:
+          regex_string: str. A regex pattern for valid ids.
+        """
+        super(init)
```

Comment:
Does this work in Python 2? I think this needs to be:

```suggestion
super(ValidateModelIdWithRegex, self).__init__()
```

Reply:
That doesn't work either way haha. I must have added this in an attempt to fix a linting thing, and didn't run the backend tests again. fixed.

Reply:
Ah, in that case it's probably because it needs args/kwargs. Perhaps update to accept and pass them along?

```python
def __init__(self, *args, **kwargs):
    super(...).__init__(*args, **kwargs)
```